### PR TITLE
Remove iree-enable-fusion-with-reduction-ops flag

### PR DESCRIPTION
**Pull request by ThomasRaoux**
This change should have no effect as the flag is already on by default

### CODE REVIEW

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
All usch PRs need to have a benchmark run on them to know if they regress performance or not. I am happy to make this option enabled by default, but it caused regression on SPIR-V IIRC.

**Comment by ThomasRaoux** [comment]
> All usch PRs need to have a benchmark run on them to know if they regress performance or not. I am happy to make this option enabled by default, but it caused regression on SPIR-V IIRC.

The flag is already on by default so this PR should make not change anything at all.

Here is the PR which turned it on by default where I did run benchmark and nothing regressed:
https://github.com/google/iree/pull/8366

**Comment by MaheshRavishankar** [comment]
> > All usch PRs need to have a benchmark run on them to know if they regress performance or not. I am happy to make this option enabled by default, but it caused regression on SPIR-V IIRC.
> 
> The flag is already on by default so this PR should make not change anything at all.
> 
> Here is the PR which turned it on by default where I did run benchmark and nothing regressed: #8366

Wasnt cc-ed on that PR, so I had no idea.

**Review by MaheshRavishankar — APPROVED** [review]
I'm fine with removing this flag, but its not always a clear win. When you fuse an elementwise operation with a reduction operation you reduce the parallelism available in favor of reducing dispatches. We anyway need a different way of controlling this.

**Comment by ThomasRaoux** [comment]
> I'm fine with removing this flag, but its not always a clear win. When you fuse an elementwise operation with a reduction operation you reduce the parallelism available in favor of reducing dispatches. We anyway need a different way of controlling this.

Of course, I believe we discussed it already. Having a flag is not the right solution though (as I thought we had agreed). Obviously we need a fusion cost model for that and several other cases.

**Comment by ThomasRaoux** [comment]
> 



> > I'm fine with removing this flag, but its not always a clear win. When you fuse an elementwise operation with a reduction operation you reduce the parallelism available in favor of reducing dispatches. We anyway need a different way of controlling this.
> 
> Of course, I believe we discussed it already. Having a flag is not the right solution though (as I thought we had agreed). Obviously we need a fusion cost model for that and several other cases.

And the reason for making this the default behavior is that we already have to turn it on for all our GPU benchmarks (GPU is really where the parallelism will matter the most) so having the flag there and off by default seems strictly negative.

**Comment by MaheshRavishankar** [comment]
> > I'm fine with removing this flag, but its not always a clear win. When you fuse an elementwise operation with a reduction operation you reduce the parallelism available in favor of reducing dispatches. We anyway need a different way of controlling this.
> 
> Of course, I believe we discussed it already. Having a flag is not the right solution though (as I thought we had agreed). Obviously we need a fusion cost model for that and several other cases.

Yeah, I am reworking the fusion part anyway. I think a basic (binary) cost model is good enough for now. This is fine to land. I was planning to enable this by default anyway, but wasnt sure what impact on perf would be. So thanks for doing this pro-actively.

File: benchmarks/TFLite/CMakeLists.txt [hunk 1]
```diff
@@ -548,7 +548,6 @@ iree_benchmark_suite(
     "GPU-Adreno"
   TRANSLATION_FLAGS
     ${ANDROID_ADRENO_GPU_TRANSLATION_FLAGS}
```
Comment (GMNGeoffrey) [inline benchmarks/TFLite/CMakeLists.txt:550]: Whoo! This means no experimental flags here and in a few places below. With our current setup, I think that means we're going to be running the same benchmarks twice now :-/ We could drop the experimental flags benchmarks entirely, although that means we're not going to have history for them if/when we want to revive them. This is another reason to decouple benchmark data collection from the plotting of the data series. @antiagainst thoughts?

**Pull request merged**